enterprise/frontend/test/metabase-enterprise/sandboxes/sandboxes.cy.spec.js

```diff
@@ -914,3 +867,51 @@ describeWithToken("formatting > sandboxes", () => {
     });
   });
 });
+
+function signInAsSandboxedUser() {
+  cy.log("**-- Logging in as sandboxed user --**");
+  cy.request("POST", "/api/session", {
+    username: sandboxed_user.email,
+    password: sandboxed_user.password,
+  });
+}
+
+/**
+ * As per definition for `PUT /graph` from `permissions.clj`:
+ *
+ * This should return the same graph, in the same format,
+ * that you got from `GET /api/permissions/graph`, with any changes made in the wherever necessary.
+ * This modified graph must correspond to the `PermissionsGraph` schema.
+ *
+ * That's why we must chain GET and PUT requests one after the other.
+ */
+
+function updatePermissionsGraph({
+  schema = {},
+  user_group = COLLECTION_GROUP,
+  database_id = 1,
+} = {}) {
+  if (typeof schema !== "object") {
+    throw new Error("`schema` must be an object!");
+  }
+
+  cy.log("**-- Fetch permissions graph --**");
+  cy.request("GET", "/api/permissions/graph", {}).then(
+    ({ body: { groups, revision } }) => {
+      // This mutates the original `groups` object => we'll pass it next to the `PUT` request
+      groups[user_group] = {
```

Comment:
Maybe `groups[user_group][database_id] = {schemas:...` instead?

If we know that both those keys are defined, that change will prevent deleting the other databases if we add those down the line. Though, I'm not sure how the endpoint works so maybe not including some database ids for a group is fine anyways.

Reply:
I could've used something from `_underscore`, but I thought this was simple enough. As for the potential external/additional database ids, I am not sure how can we override those? The function accepts `database_id` parameter so it should be pretty straightforward and obvious which database schema we're updating, right?

If I'm not mistaken, the following scenario should be valid:
```javascript
updatePermissionsGraph({database_id: 1, schema: {...}})

// then immediately after that:
updatePermissionsGraph({database_id: 2, schema: {...}})
```

If I'm not mistaken it works cumulatively?

P.S. In any case, this function is (so far) localized to this file only. We can monitor it and add update it as needed.

Thank you for taking the time to review this.

Reply:
I'm assuming we need to post up the entire graph each time, but I haven't tested it. The way this is written now, if there are multiple dbs, the others will get dropped within `groups[user_group]`. 

`{[group_id]: {1: {schemas:...}, 2: {schemas:...}}}` => `{[group_id]: {1: {schemas:...}}}` if we set `groups[group_id] = {1:  {schemas:...}}`

Merging and waiting for any issues is a good call though 👍